internal/character/danhengimbibitorlunae/eidolon.go

```diff
@@ -0,0 +1,30 @@
+package danhengimbibitorlunae
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	E6Effect = "danhengimbibitorlunae-e6effect" // MAvatar_DanHengIL_00_Rank06_ImaginaryPenetrate
+)
+
+// E1 increase talent max stack by 4 and gain 1 extra stack when hit
+// E2 100% forward and 1 more ult stack
+// E4 skill buff 1 more turn
+// E6 20% imaginary pen for attack3 when ally use ult, max 3 stack
+
+func init() {
+	// imaginary pen for attack3,change by ally ult count
+	modifier.Register(E6Effect, modifier.Config{
+		StatusType: model.StatusType_STATUS_BUFF,
```

Comment:
E6 effect is a modifier that listen to OnBeforeHitAll and check if the attacks is lvl 3 then adds the res pen to that attack
```suggestion
		StatusType: model.StatusType_STATUS_BUFF,
		MaxCount: 3,
		CountAddWhenStack: 1,
		Stacking: modifier.ReplaceBySource,
		Listeners: modifier.Listeners{
			OnBeforeHitAll: E6PenOnBeforeHitAll,
			OnAfterAttack: E6PenOnAfterAttack,
		}
```
The beforehit function would look something like this.
```
func E6PenOnBeforeHitAll(mod *modifier.Instance, e event.HitStart) {
	if //check here if attack level is 3 {
		amt := mod.Count()*mod.State().(float64)
		e.Hit.Attacker.AddProperty(name, prop.ImaginaryPEN, amt)
	}
}
```
The afterattack function would look something like this.
```
func E6PenOnAfterAttack(mod *modifier.Instance, e event.AttackEnd) {
	if //check here if attack level is 3 {
		//remove E6Effect mod
	}
}
```